### PR TITLE
now able to do basic player/guild search switching

### PR DESCRIPTION
**Pull request by OliverRadini**
Added search to front page, functionality currently fairly basic, but able to switch between guild and player searches. Will continue working on this as it's not completely finished, but makes sense to put in a pull request now as there hasn't been anything for a while, and I don't want to get too far ahead of master.

### CODE REVIEW

**Comment by burfo** [comment]
@Donutttt , you should run the tests before committing.  `npm run test`

**Comment by OliverRadini** [comment]
@burfo tests won't run on my local machine anymore, saying it can't start chrome. The errors on Travis don't really make a whole lot of sense.

**Comment by OliverRadini** [comment]
no idea with this one anymore; it's failing on GuildSearchService not having a provider, but from what I can see it does; it seems to be setup in the exact same way as PlayerSearchService. Any ideas on how to fix it please let me know

**Comment by OliverRadini** [comment]
Just to be clear, I'm out of ideas on fixing this. It's only failing because of how the tests are setup, as far as I can see, but I can't make it work. If someone can get this to pass the tests then I'll get back to working on stuff, can't spend any more time on these tests atm.

**Comment by burfo** [comment]
@Donutttt, fix this invalid path in home.component.spec.ts and the error will go away:
`import {GuildSearchService} from '../shared/player-search.service';`

However, then you'll see a subsequent error for HomeComponent:

> Error in :0:0 caused by: $ is not defined

That error happens because we're not loading jQuery through a package but instead directly inside index.html.  Karma doesn't know about index.html.  Thus, you need to define the jquery source file into karma.conf.js ([reference](http://stackoverflow.com/questions/28973220/karma-uncaught-referenceerror-is-not-defined)).

**Comment by OliverRadini** [comment]
@burfo thanks for taking a look at that, my guess was it was something stupid but I couldn't see it! I can work on this later. I'll take a look at the jquery part as well

**Comment by OliverRadini** [comment]
now working

File: herald/src/app/+guild-profile/shared/guild-profile.service.ts [hunk 1]
```diff
@@ -22,8 +22,9 @@ export class GuildProfileService {
   }
 
   private getGuildProfileFromFile(name: string) {
+    console.log('making call from guild profile service for ' + name);
     let fileName = name.replace(/ /g, '-');
-    return this.http.get(`/assets/data/guilds/${fileName}.json`)
+    return this.http.get(`https://www2.uthgard.net/herald/api/guild/${fileName}`)
```
Comment (andir) [inline herald/src/app/+guild-profile/shared/guild-profile.service.ts]: Knowing that I'm gonna be a pain:

Didn't we introduce a single `class` to carry all those variables?

**Review by andir — APPROVED** [review]
LGTM besides that one comment regarding the file url

File: herald/src/app/+guild-profile/shared/guild-profile.service.ts (hunk 1 above)
Comment (OliverRadini) [reply herald/src/app/+guild-profile/shared/guild-profile.service.ts]: Yes, and in fact I think I've used it elsewhere, I can get that changed later

Comment (OliverRadini) [reply herald/src/app/+guild-profile/shared/guild-profile.service.ts]: This is now updated

**Pull request merged**